src/build123d/mesher.py

```diff
@@ -551,3 +553,8 @@ def write(self, file_name: PathLike | str | bytes):
             raise ValueError(f"Unknown file format {output_file_extension}")
         writer = self.model.QueryWriter(output_file_extension[1:])
         writer.WriteToFile(file_name)
+
+    def write_stream(self, stream: BytesIO, file_type: str):
```

Comment:
@gumyr do you have any opinion on the naming for this new `Mesher.write_stream` method? The backstory for why I chose to do this is that the `Mesher.write` method infers the file format (STL or 3MF) from the output file extension, which is not applicable to BytesIO.